src/Connections/GithubRepo.ts

```diff
@@ -1260,12 +1266,72 @@ export class GitHubRepoConnection extends CommandConnection<GitHubRepoConnection
-        await this.intent.sendEvent(this.roomId, {
+
+        let extraThreadContent: {plain: string, html: string}|null = null;
+
+        if (workflowRun.conclusion === "failure") {
+            let jobs;
+            try {
+                jobs = await this.githubInstance.getOctokitForRepo(
+                    event.repository.owner.login,
+                    event.repository.name,
+                ).actions.listJobsForWorkflowRunAttempt({
+                    owner: event.repository.owner.login,
+                    repo: event.repository.name,
+                    run_id: event.workflow_run.id,
+                    attempt_number: event.workflow_run.run_attempt,
+                });
+
+                extraThreadContent = {
+                    plain: "#### Workflow log \n\n",
+                    html: "<h4> Workflow log </h4><ul>",
+                };
+                
+                for (const job of jobs.data.jobs) {
+                    const jobConclusion = job.conclusion ? WORKFLOW_JOB_TO_NOTICE[job.conclusion] : "No conclusion";
+                    extraThreadContent.plain += `  - Job \`${job.name}\` - ${jobConclusion}\n`;
+                    extraThreadContent.html += `<li> Job <code>${job.name}</code> ${jobConclusion}</li><ul>`;
+                    if (job.conclusion !== "success") {
+                        for (const step of job.steps || []) {
+                            const conclusion = step.conclusion ? WORKFLOW_JOB_TO_NOTICE[step.conclusion] : "No conclusion";
+                            extraThreadContent.plain += `    - <strong>${conclusion}</strong> ${step.name} \n`;
+                            extraThreadContent.html += `<li><strong>${conclusion}</strong> ${step.name} </li>`;
+                        }
+                    }
+                    extraThreadContent.html += `</ul>`;
+                }
+                extraThreadContent.html += `</ul>`;
+            } catch (ex) {
+                log.warn(`Failed to fetch extra workflow info`, ex);
+            }
+        }
+
+        const matrixEvent = await this.intent.sendEvent(this.roomId, {
             msgtype: "m.notice",
             body: content,
             formatted_body: md.render(content),
             format: "org.matrix.custom.html",
+            ...FormatUtil.getPartialBodyForGithubRepo(event.repository),
         });
+
+        if (extraThreadContent) {
+            this.intent.sendEvent(this.roomId, {
+                "m.relates_to": {
+                    rel_type: "m.thread",
+                    event_id: matrixEvent,
+                    // Needed to prevent clients from showing these as actual replies
+                    is_falling_back: true,
+                    "m.in_reply_to": {
+                        event_id: matrixEvent,
+                    }
```

Comment:
Do you really want this to be a reply too or just part of the thread?

Reply:
I think we want the fallback to be a reply?

Reply:
Well then, I had fully forgotten this was a thing. 👍